### PR TITLE
Add search functionality on collections, Features, Templates

### PR DESCRIPTION
**Pull request by bamurtaugh**
This pull request adds search functionality for collections, Templates, and Features.

- Added an input field for searching tables

- Implemented JavaScript logic to filter the table rows based on the search input

These changes improve the user experience by allowing users to easily search and find specific items in each table.

### CODE REVIEW

File: collections.html [hunk 1]
```diff
@@ -17,17 +17,44 @@ <h1 style="margin-left: auto;margin-right: auto;">Collections</h1>
...
-<tr>
-    <td class="tg-0lax">{{ c.name }}</td>
-    <td class="tg-0lax"><a rel="nofollow" href="{{ c.contact | strip_html  }}">{{ c.maintainer | strip_html }}</a></td>
-    <td class="tg-0lax"><a rel="nofollow" href="{{ c.repository | strip_html  }}">{{ c.repository | strip_html }}</a>
-    </td>
-</tr>
-{% endfor %}
\ No newline at end of file
+
+<input type="text" id="searchInput" placeholder="Search">
```
Comment (samruddhikhandale) [inline collections.html:21]: wow! 🎉 

This is super cool!

image

File: collections.html [hunk 2]
```diff
@@ -17,17 +17,44 @@ <h1 style="margin-left: auto;margin-right: auto;">Collections</h1>
...
+
+<table id="collectionTable" style="border-collapse: collapse; border: 1px solid #ccc;">
```
Comment (samruddhikhandale) [inline collections.html]: If possible, can we add similar search functionality for https://containers.dev/features and https://containers.dev/templates? That would be pretty awesome!

File: collections.html [hunk 3]
```diff
@@ -17,17 +17,44 @@ <h1 style="margin-left: auto;margin-right: auto;">Collections</h1>
...
+    <tr>
+        <td class="tg-0lax"><b>Name</b></td>
+        <td class="tg-0lax"><b>Maintainer</b></td>
+        <td class="tg-0lax"><b>Repository</b></td>
+    </tr>
+
```
Comment (samruddhikhandale) [inline collections.html:32]: image

I don't have a super strong opinion, but I think I like how we currently have the border lines of the table (for each row and columns) highlighted.

**Review by samruddhikhandale — COMMENTED** [review]
Thank you so much for adding these changes, this will be so helpful for folks! 🎉 
Left few comments.

**Comment by bamurtaugh** [comment]
Ah sorry for any confusion! This was one of my demo scenarios during Ignite and a sample PR - I'll close this 😄

**Comment by samruddhikhandale** [comment]
Ah! 🤦‍♀️ 

@bamurtaugh FYI, this is really a good PR, adding a `search` functionality would definitely be helpful for folks.
Let me know if I could help to get this merged in.

**Comment by bamurtaugh** [comment]
Thanks so much for the feedback @samruddhikhandale and I agree we should reopen, especially as part of https://github.com/devcontainers/spec/issues/405! Let me address your feedback and then re-request review.

File: collections.html (hunk 1 above)
Comment (bamurtaugh) [reply collections.html:21]: Thank you!!

File: collections.html (hunk 2 above)
Comment (bamurtaugh) [reply collections.html]: Great feedback. I just tried adding the same logic - do you know the best way to test this? When I run the site locally, the collections data loads for me, so I can actually test searching on that page.

File: collections.html (hunk 3 above)
Comment (bamurtaugh) [reply collections.html:32]: Updated the formatting! Let me know if it looks good to you too.

Comment (samruddhikhandale) [reply collections.html:32]: Looks great!

**Review by samruddhikhandale — APPROVED** [review]
Looks great, thank you! ⚡

**Pull request merged**